eva-server/src/test/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,181 @@
+    @Autowired
+    private ApplicationContext applicationContext;
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Before
+    public void setUp() throws Exception { }
+
+    @Test
+    public void rootTestGetVariantsByVariantId() throws URISyntaxException {
+        List<VariantWithSamplesAndAnnotation> variantWithSamplesAndAnnotations = variantWsHelper("20:60100:A:T");
+        assertTrue(variantWithSamplesAndAnnotations.size() > 0);
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getSourceEntries().size() == 0);
+        assertNull(variantWithSamplesAndAnnotations.get(0).getAnnotation());
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getIds().size() > 0);
+    }
+
+    private List<VariantWithSamplesAndAnnotation> variantWsHelper(String testVariantId) {
+        String url = "/v2/variants/" + testVariantId + "?species=mmusculus_grcm38";
+        return WSTestHelpers.testRestTemplateHelper(url, restTemplate);
+    }
+
+    @Test
+    public void rootTestGetVariantsByNonExistingVariantId() throws URISyntaxException {
+        assertEquals(0, variantWsHelper("10:0:A:T").size());
+    }
+
+    @Test
+    public void rootTestForError() throws URISyntaxException {
+        String url;
+        url = "/v2/variants/13:32889669:C:T?species=";
+        assertEquals("Please specify a species", testForErrorHelper(url));
+        url = "/v2/variants/13:32889669:C:T?species=mmusculus_grcm38&annot-vep-version=1";
+        assertEquals("Please specify either both annotation VEP version and annotation VEP cache version, " +
+                "or neither", testForErrorHelper(url));
+    }
+
+    private String testForErrorHelper(String url) {
+        return WSTestHelpers.testRestTemplateHelperForError(url, restTemplate);
+    }
+
+    @Test
+    public void annotationEndPointTestExisting() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T/annotations?species=mmusculus_grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
+        assertFalse(annotations.getBody().getResponse().get(0).getResult().get(0).getChromosome().isEmpty());
+    }
+
+    @Test
+    public void annotationEndPointTestNonExisting() throws URISyntaxException {
+        String url = "/v2/variants/100:0:C:T/annotations?species=mmusculus_grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
```

Comment:
In others of our APIs, when an endpoint returns no results, the HTTP code is 404. Applies to this class and `VariantWSServerV2Test`.

Reply:
I will change it :+1: